domain-model/src/main/java/md/maib/retail/Model/Campaign/CampaignMetaInfo.java

```diff
@@ -0,0 +1,35 @@
+package md.maib.retail.Model.Campaign;
+
+import lombok.ToString;
+
+import java.util.Map;
+import java.util.Objects;
+
+import static java.util.Objects.requireNonNull;
+
+@ToString
+public class CampaignMetaInfo {
+
```

Comment:
record automatically realize toString, equals & hashCode methods

public record CampaignMetaInfo(Map<String, Object> properties) {

    public CampaignMetaInfo {
        notNull(properties, "properties");
    }
}

Reply:
like this?
package md.maib.retail.Model.Campaign;


import java.util.Map;
import static java.util.Objects.requireNonNull;

public record CampaignMetaInfo(Map<String, Object> properties) {

    public CampaignMetaInfo{
        requireNonNull(properties, "Properties must not be null.");    }
}

i should to make and CampaignId as a record ?